src/overlays/actors/ovl_Obj_Kendo_Kanban/z_obj_kendo_kanban.c

```diff
@@ -217,238 +233,251 @@ void ObjKendoKanban_Destroy(Actor* thisx, PlayState* play) {
 }
 
-void func_80B65CE0(ObjKendoKanban* this, PlayState* play) {
+void ObjKendoKanban_Aerial(ObjKendoKanban* this, PlayState* play) {
     this->actor.velocity.y += this->actor.gravity;
     Actor_UpdatePos(&this->actor);
-    this->unk_302 += this->unk_304;
-    func_80B65DA8(this, play);
+    this->rotationAngle += this->rotationVelocity;
+    ObjKendoKanban_HandlePhysics(this, play);
     if (this->actor.world.pos.y < -200.0f) {
         this->actor.world.pos.y = -200.0f;
     }
 }
 
-void func_80B65D54(ObjKendoKanban* this) {
-    this->actionFunc = func_80B65D68;
+void ObjKendoKanban_SetupSettled(ObjKendoKanban* this) {
+    this->actionFunc = ObjKendoKanban_Settled;
 }
 
-void func_80B65D68(ObjKendoKanban* this, PlayState* play) {
+void ObjKendoKanban_Settled(ObjKendoKanban* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
 
-    if (func_80B6618C(this, play) == 1) {
-        player->unk_AC0 = 700.0f;
+    if (ObjKendoKanban_IsPlayerOnTop(this, play) == true) {
+        player->displacementY = 700.0f;
     }
 }
 
-void func_80B65DA8(ObjKendoKanban* this, PlayState* play) {
-    Vec3f sp5C;
+void ObjKendoKanban_HandlePhysics(ObjKendoKanban* this, PlayState* play) {
+    Vec3f rootCornerPos = this->cornerPos[0];
     s32 pad[2];
-    Vec3f sp48;
-    s32 pad2;
-    s32 index = 0;
+    Vec3f vecCenterOut = this->actor.world.pos;
+    const s16 MAX_BOUNCE_COUNT = 7;
+    s32 idxRootCornerPos = 0;
     s32 i;
-    f32 sp38;
-
-    sp5C = this->unk_26C[0];
-    sp48 = this->actor.world.pos;
-
-    sp48.x -= this->unk_2D8.x;
-    sp48.y -= this->unk_2D8.y;
-    sp48.z -= this->unk_2D8.z;
-
-    sp38 = (this->unk_2F0.x * sp48.z) + (this->unk_2F0.z * -sp48.x);
-
-    if (sp38 < 0.0f) {
-        this->unk_304 += 0x64;
+    f32 verticalScalar;
+
+    // Calculate an affect on the rotation from gravity.
+    vecCenterOut.x -= this->centerPos.x;
+    vecCenterOut.y -= this->centerPos.y;
+    vecCenterOut.z -= this->centerPos.z;
+    verticalScalar = (this->rotationalAxis.x * vecCenterOut.z) + (this->rotationalAxis.z * -vecCenterOut.x);
+    if (verticalScalar < 0.0f) {
+        this->rotationVelocity += 0x64;
     } else {
-        this->unk_304 -= 0x64;
+        this->rotationVelocity -= 0x64;
     }
 
-    for (i = 0; i < ARRAY_COUNT(this->unk_26C); i++) {
-        if (this->unk_26C[i].y < sp5C.y) {
-            sp5C = this->unk_26C[i];
-            index = i;
+    // Find the lowest point
+    for (i = 0; i < ARRAY_COUNT(this->cornerPos); i++) {
+        if (this->cornerPos[i].y < rootCornerPos.y) {
+            rootCornerPos = this->cornerPos[i];
+            idxRootCornerPos = i;
         }
     }
 
-    if (index != this->unk_2FC) {
-        this->unk_300 = 1;
-        this->unk_2FC = index;
-        this->unk_2E4 = this->unk_29C[index];
+    // When the lowest point changes, re-initialize the actor position to that point
+    if (idxRootCornerPos != this->idxLastRootCornerPos) {
+        this->bHasNewRootCornerPos = true;
```

Comment:
Out of curiosity, what is the `b` for?

Reply:
My style consistency is leaking. Some code I interact with makes heavy use of hungarian notation, and that slipped through. It notes that it's used as a bool. "Is"/"has" works perfectly fine for that and is more consistent, so I can drop the "b" no problem.